lib/temporal/client/retryer.rb

```diff
@@ -0,0 +1,29 @@
+module Temporal
+  module Client
+    module Retryer
+      INITIAL_INTERVAL_S = 0.2
+      MAX_INTERVAL_S = 6.0
+      BACKOFF_COEFFICIENT = 1.2
+
+      # Used for backoff retries in certain cases when calling temporal server.
+      # metadata_hash: for logging, pass in metadata.to_h
+      def self.retry_for(give_up_after_s, retry_message:, metadata_hash:, &block)
```

Comment:
Depending retryer on `metadata` sounds very wrong — what if we then decide to use it somewhere where metadata is not available, e.g. when polling for tasks or resetting a workflow

Reply:
https://github.com/kamui/retriable or https://github.com/nfedyashev/retryable for inspiration